tests/unit/test_artifactory_path.py

```diff
@@ -1158,5 +1161,229 @@ def test_from_aql_file(self):
+            )
+            self.assertEqual(
+                rsps.calls[2].request.url, f"{self.groups_request_url}/group_2"
+            )
+
+    def test_get_groups_lazy(self):
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.GET, self.groups_request_url, json=self.groups, status=200
+            )
+
+            results = self.arti.get_groups(lazy=True)
+
+            for group in results:
+                self.assertIsInstance(group, Group)
+            self.assertEqual(results[0].name, "group_1")
+            self.assertNotEqual(results[0].realm, "internal")
+            self.assertEqual(results[1].name, "group_2")
+            self.assertNotEqual(results[1].realm, "internal")
+
+            self.assertEqual(len(rsps.calls), 1)
+            self.assertEqual(rsps.calls[0].request.url, self.groups_request_url)
+
+    def test_get_projects(self):
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.GET, self.projects_request_url, json=self.projects, status=200
+            )
+
+            rsps.add(
+                responses.GET,
+                f"{self.projects_request_url}/project_key_1",
+                json=self.project_1,
+                status=200,
+            )
+            rsps.add(
+                responses.GET,
+                f"{self.projects_request_url}/project_key_2",
+                json=self.project_2,
+                status=200,
+            )
+
+            results = self.arti.get_projects(lazy=False)
+
+            for project in results:
+                self.assertIsInstance(project, Project)
+            self.assertEqual(results[0].project_key, "project_key_1")
+            self.assertEqual(results[0].description, "description_1")
+            self.assertEqual(results[1].project_key, "project_key_2")
+            self.assertEqual(results[1].description, "description_2")
+
+            self.assertEqual(len(rsps.calls), 3)
+            self.assertEqual(rsps.calls[0].request.url, self.projects_request_url)
+            self.assertEqual(
+                rsps.calls[1].request.url, f"{self.projects_request_url}/project_key_1"
+            )
+            self.assertEqual(
+                rsps.calls[2].request.url, f"{self.projects_request_url}/project_key_2"
+            )
+
+    def test_get_projects_lazy(self):
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.GET, self.projects_request_url, json=self.projects, status=200
+            )
+
+            results = self.arti.get_projects(lazy=True)
+
+            for project in results:
+                self.assertIsInstance(project, Project)
+            self.assertEqual(results[0].project_key, "project_key_1")
+            self.assertNotEqual(results[0].description, "description_1")
+            self.assertEqual(results[1].project_key, "project_key_2")
+            self.assertNotEqual(results[1].description, "description_2")
```

Comment:
why do we do a not Equal?
may be worth to check that it is kept as default class value? eg according to [L1315](https://github.com/devopshq/artifactory/blob/a14e7bf639c0a389bf104bdd9f25d32715113f13/dohq_artifactory/admin.py#L1315)
```suggestion
            self.assertFalse(results[1].description)
```

Reply:
how about self.assertEqual(results[1].description, "")?

Reply:
usually in the code you will run `if not var.description`, but for this particular case to check that bool or other value was not leaked that makes sense
go ahead

Reply:
Thanks. Will do it tomorrow

Reply:
Hi @beliaev-maksim 

This issue is fixed. Could you kindly review and approve this PR?